src/Utilities/Compiler/Extensions/CompilationExtensions.cs

```diff
@@ -59,5 +61,15 @@ internal static bool TargetsDotNetFramework(this Compilation compilation)
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a value indicating whether the project of the compilation is a Web SDK project,
+        /// based on MSBuild property <see cref="MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb"/>.
+        /// </summary>
+        internal static bool IsWebProject(this Compilation compilation, AnalyzerOptions options, CancellationToken cancellationToken)
+        {
+            var usingMicrosoftNETSdkWeb = options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.UsingMicrosoftNETSdkWeb, compilation, cancellationToken);
```

Comment:
Is this property valid only for aspnet core or also regular aspnet? If it's valid only for aspnet core, do we really want to get rid of the old exclusions? That would mean raising FPs for "old" projects.

Reply:
It will be valid for all SDK based aspnet projects. @pranavkm Is there any MSBuild property that is set for regular aspnet projects that we can use?

Reply:
> Is there any MSBuild property that is set for non-SDK based aspnet projects that we can use?

Not that I know of.

Reply:
I looked around and the only way I could identify is by looking at `ProjectTypeGuids` property.

```csharp
public const string WebAppProjectGuidString = "349C5851-65DF-11DA-9384-00065B846F21";
public const string WebSiteProjectGuidString = "E24C65DC-7377-472B-9ABA-BC803B73C61A";
```

I'll can add these property value checks unless we find a better way.